src/Impl/CopyGitLink.Shared/Core/GitOnlineServices/GitHubnLab.cs

```diff
@@ -123,29 +140,40 @@ private bool TryParseGitUri(string repositoryUriString, out IDictionary<string,
 
             properties = new Dictionary<string, string>();
 
-            // Detect if the host corresponds to GitHub.
+            // The host can be github, gitlab or a Self-Managed version of both.
             // It will be in the form of 
-            // https://github.com/{org or user}/{repo name}.git
-            if (string.Equals(repositoryUri.Host, "github.com", StringComparison.OrdinalIgnoreCase)
-                && repositoryUri.Segments.Length == 3)
+            // https://{github|gitlab|Self-Managed}.{extension}/{org or user}/{repo name}.git
+            // Must have .git uri ending
+            var repositoryNameIndex = repositoryUri.Segments.Length - 1;
+            if (repositoryUri.Segments[repositoryNameIndex].IndexOf(RemoteGitEnding, StringComparison.Ordinal) > 0)
             {
-                // Must have .git uri ending
-                if (repositoryUri.Segments[2].IndexOf(RemoteGitEnding, StringComparison.Ordinal) > 0)
-                {
-                    // Trims the .git suffix
-                    properties[Repository] = repositoryUri.Segments[2].Substring(0, repositoryUri.Segments[2].Length - 4);
-                }
-                else
-                {
-                    return false;
-                }
-
-                properties[Organization] = repositoryUri.Segments[1].TrimEnd('/');
+                // Trims the .git suffix
+                properties[Host] = repositoryUri.Host;
+                properties[Repository] = repositoryUri.Segments[repositoryNameIndex].Substring(0, repositoryUri.Segments[repositoryNameIndex].Length - 4);
+                var organizationInfo = repositoryUri.Segments.Skip(1).Take(repositoryUri.Segments.Length - 2);
+                properties[Organization] = string.Join("", organizationInfo).TrimEnd('/');
+                
                 return true;
             }
+            else
+            {
+                return false;
+            }
+        }
 
-            properties = null;
-            return false;
+
+        /// <summary>
+        /// convert github, gitlab or self-managed version repository SSH url to http url
+        /// </summary>
+        /// <param name="repositoryUriString"></param>
+        /// <returns></returns>
+        private string ConvertSshUrlToHttpUrl(string repositoryUriString)
+        {
+            if (repositoryUriString.Length > 4)
+            {
+                return $"http://{repositoryUriString.Substring(4).Replace(':', '/')}";
```

Comment:
The question will remain the same if i use https instead of http.

Reply:
I use `http` because in general when a user already visited an HTTPS version, browser will automatically redirects http to https version. But yes i can replace it with `https` .